internal/job/checkout.go

```diff
@@ -491,6 +492,15 @@ func (e *Executor) defaultCheckoutPhase(ctx context.Context) error {
 
 	gitFetchFlags := e.GitFetchFlags
 
+	if e.PullRequest != "false" && strings.Contains(e.PipelineProvider, "bitbucket") {
```

Comment:
regardless of that: i wonder if it might be best to just remove this conditional, and always rev-parse the commit - `git rev-parse <some full commit sha` is kind of an identity function, and as far as i know it doesn't have any major performance concerns.

Reply:
That's ok for me, but would `o.Commit` always contain something that can be _rev-parsed_, I'd guess it will be the case? 
Then it's more or less what the `resolveCommit` method does but that's applied after the checkout is completed and updates the `BUILDKITE_COMMIT` environment variable as well of course.

So just:
```go
if o, err := e.shell.RunAndCapture(ctx, "git", "rev-parse", e.Commit); err != nil {
  return fmt.Errorf("rev-parse %w", err)
} else {
  e.Commit = o
}
```

If `e.Commit` is not guaranteed to be _rev-parsed_ we might just log the error and try to continue?
```go
if o, err := e.shell.RunAndCapture(ctx, "git", "rev-parse", e.Commit); err != nil { 
  e.shell.Warningf("failed to rev-parse %s, %w", e.Commit, err)
} else {
  e.Commit = o
}
```

Reply:
@moskyb could we decide on a path forward? I'll update the code accordingly so that we can get this PR merged and released (if it's the correct solution) 😊

Reply:
i'm pretty sure that `o.Commit` should always be rev-parsable, but just in case i like the second option - attempting a rev-parse and just logging if it doesn't work.